src/com/rcv/Tabulator.java

```diff
@@ -552,4 +530,16 @@ private int numCandidates() {
     }
     return num;
   }
+
+  public void doAudit(List<CastVoteRecord> castVoteRecords) {
+    for(CastVoteRecord cvr : castVoteRecords) {
+      String auditString = String.format("%s:",cvr.mBallotID);
```

Comment:
Use a StringBuilder instead. Because Strings are immutable in Java, building up a long string through concatenation is very inefficient -- it will instantiate a new object each time you concatenate.

Reply:
will do